### ISSUE TITLE
Crash on dropping in same area multiple times

### ISSUE THREAD

**Issue by iam-peter** [githubuser0xFFFF/Qt-Advanced-Docking-System#131]
When dropping the same `DockWidget` on the same drop area multiple times the demo application crashes. The gif showing how a user drops the `DockWidget` *Label 5* twice on the bottom drop area. The application is crashing on release of the mouse at the second drop.

![output](https://user-images.githubusercontent.com/7542157/76404644-13ac1180-6387-11ea-94e4-8c2f72050e54.gif)

This was tested with an up-to-date version of the demo application, compiled with Qt 5.12, 5.13 and 5.15 on Windows and Ubuntu.

```bash
#0  0x00007ffff79bcf00 in QSplitter::indexOf(QWidget*) const (this=0x0, w=0x555555a25380) at widgets/qsplitter.cpp:1211
#1  0x00007ffff7f7808b in ads::DockContainerWidgetPrivate::moveToNewSection(QWidget*, ads::CDockAreaWidget*, ads::DockWidgetArea)
    (this=0x555555966a40, Widget=0x555555a25380, TargetArea=0x555555a25380, area=ads::BottomDockWidgetArea) at ../../Qt-Advanced-Docking-System/src/DockContainerWidget.cpp:636
#2  0x00007ffff7f7b05c in ads::CDockContainerWidget::dropWidget(QWidget*, QPoint const&) (this=0x555555966800, Widget=0x555555a25380, TargetPos=...)
    at ../../Qt-Advanced-Docking-System/src/DockContainerWidget.cpp:1461
#3  0x00007ffff7f90d70 in ads::CFloatingDragPreview::finishDragging() (this=0x5555559c2b80) at ../../Qt-Advanced-Docking-System/src/FloatingDragPreview.cpp:294
#4  0x00007ffff7f8baf4 in ads::CDockWidgetTab::mouseReleaseEvent(QMouseEvent*) (this=0x555555a308e0, ev=0x7fffffffd220) at ../../Qt-Advanced-Docking-System/src/DockWidgetTab.cpp:332
#5  0x00007ffff7867258 in QWidget::event(QEvent*) (this=this@entry=0x555555a308e0, event=event@entry=0x7fffffffd220) at kernel/qwidget.cpp:9324
#6  0x00007ffff790d30e in QFrame::event(QEvent*) (this=0x555555a308e0, e=0x7fffffffd220) at widgets/qframe.cpp:550
#7  0x00007ffff7f8ccf2 in ads::CDockWidgetTab::event(QEvent*) (this=0x555555a308e0, e=0x7fffffffd220) at ../../Qt-Advanced-Docking-System/src/DockWidgetTab.cpp:618
#8  0x00007ffff7828e9c in QApplicationPrivate::notify_helper(QObject*, QEvent*) (this=<optimized out>, receiver=0x555555a308e0, e=0x7fffffffd220) at kernel/qapplication.cpp:3740
#9  0x00007ffff783057f in QApplication::notify(QObject*, QEvent*) (this=<optimized out>, receiver=0x555555807130, e=0x7fffffffd4b0) at kernel/qapplication.cpp:3200
#10 0x00007ffff694dea8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (receiver=0x555555807130, event=0x7fffffffd4b0) at kernel/qcoreapplication.cpp:1065
#11 0x00007ffff694e06e in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) (receiver=receiver@entry=0x555555807130, event=event@entry=0x7fffffffd4b0)
    at kernel/qcoreapplication.cpp:1472
#12 0x00007ffff782f4ba in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool)
    (receiver=receiver@entry=0x555555807130, event=event@entry=0x7fffffffd4b0, alienWidget=alienWidget@entry=0x555555a45830, nativeWidget=0x7fffffffdcb0, buttonDown=buttonDown@entry=0x7ffff7f1a950 <qt_button_down>, lastMouseReceiver=..., spontaneous=true, onlyDispatchEnterLeave=false) at kernel/qapplication.cpp:2686
#13 0x00007ffff7880d61 in QWidgetWindow::handleMouseEvent(QMouseEvent*) (this=this@entry=0x555555c2ede0, event=event@entry=0x7fffffffd890) at kernel/qwidgetwindow.cpp:663
#14 0x00007ffff78839fb in QWidgetWindow::event(QEvent*) (this=0x555555c2ede0, event=0x7fffffffd890) at kernel/qwidgetwindow.cpp:281
#15 0x00007ffff7828e9c in QApplicationPrivate::notify_helper(QObject*, QEvent*) (this=<optimized out>, receiver=0x555555c2ede0, e=0x7fffffffd890) at kernel/qapplication.cpp:3740
#16 0x00007ffff782ffa0 in QApplication::notify(QObject*, QEvent*) (this=0x7fffffffdc80, receiver=0x555555c2ede0, e=0x7fffffffd890) at kernel/qapplication.cpp:3486
#17 0x00007ffff694dea8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (receiver=0x555555c2ede0, event=0x7fffffffd890) at kernel/qcoreapplication.cpp:1065
#18 0x00007ffff694e06e in QCoreApplication::sendSpontaneousEvent(QObject*, QEvent*) (receiver=receiver@entry=0x555555c2ede0, event=event@entry=0x7fffffffd890)
    at kernel/qcoreapplication.cpp:1472
#19 0x00007ffff6fd4733 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) (e=0x55555599db90) at kernel/qguiapplication.cpp:2111
#20 0x00007ffff6fd5ce5 in QGuiApplicationPrivate::processWindowSystemEvent(QWindowSystemInterfacePrivate::WindowSystemEvent*) (e=e@entry=0x55555599db90) at kernel/qguiapplication.cpp:1846
#21 0x00007ffff6fb1a0b in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) (flags=..., flags@entry=...) at kernel/qwindowsysteminterface.cpp:1148
#22 0x00007ffff2f2cfaa in xcbSourceDispatch(GSource*, GSourceFunc, gpointer) (source=<optimized out>) at qxcbeventdispatcher.cpp:105
#23 0x00007ffff3d7a84d in g_main_context_dispatch () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#24 0x00007ffff3d7aad0 in  () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#25 0x00007ffff3d7ab73 in g_main_context_iteration () at /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
#26 0x00007ffff69a618f in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (this=0x55555562b440, flags=...) at kernel/qeventdispatcher_glib.cpp:423
#27 0x00007ffff694c902 in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (this=this@entry=0x7fffffffdc00, flags=..., flags@entry=...) at kernel/qeventloop.cpp:225
#28 0x00007ffff6955260 in QCoreApplication::exec() () at kernel/qcoreapplication.cpp:1373
#29 0x000055555555ecfd in main(int, char**) (argc=1, argv=0x7fffffffdde8) at ../../Qt-Advanced-Docking-System/demo/main.cpp:57
```

**Comment by githubuser0xFFFF** [comment]
Hi Peter, thank you for the bug report. I can reproduce this here and will fix it now.

**Comment by githubuser0xFFFF** [comment]
Fixed this issue with commit 6c687d28def5c24f3eeb67c9e13d5cfab40cebd0

**Comment by iam-peter** [comment]
Thanks for fixing the issue.
